src/Assignments/SimpleProblemPage.tsx

```diff
@@ -48,9 +48,14 @@ export const SimpleProblemPage: React.FC<SimpleProblemPageProps> = () => {
     const [alert, setAlert] = useMUIAlertState();
     const [confirmationParameters, setConfirmationParameters] = useState<ConfirmationModalProps>(DEFAULT_CONFIRMATION_PARAMETERS);
     const [openDrawer, setOpenDrawer] = useState<boolean>(false);
+    const [smaHasNoVersions, setSmaHasNoVersions] = useState<boolean>(false);
     const {course, users} = useCourseContext();
     const noAlert = useRef<IMUIAlertModalState>({severity: 'info', message: ''});
 
+    useEffect(() => {
+        setSmaHasNoVersions(false);
+    }, [setSmaHasNoVersions, selectedProblemId]);
```

Comment:
what purpose does it serve to watch the function: `setSmaHasNoVersions`?

Reply:
I'm not sure if it matters, I thought it yelled at you if you used things that were not watched 🤣